### PR TITLE
[REVIEW] build.sh option to disable nvtx

### PR DESCRIPTION
**Pull request by jdye64**
By specifying `-x' flag. Closes: #3524

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -31,6 +31,7 @@ HELP="$0 [clean] [libcudf] [cudf] [dask_cudf] [benchmarks] [-v] [-g] [-n] [-h]
    -v           - verbose build mode
    -g           - build for debug
    -n           - no install step
+   -x           - do not build nvtx
```
Comment (harrism) [inline build.sh]: I don't like `-x` for this. How about `--disable_nvtx`?

File: conda/recipes/libcudf/build.sh [hunk 2]
```diff
@@ -7,4 +7,4 @@ printenv
 # Cleanup local git
 git clean -xdf
 # build libcudf with verbose output
-./build.sh -v libcudf --allgpuarch
+./build.sh -v -x libcudf --allgpuarch
```
Comment (harrism) [inline conda/recipes/libcudf/build.sh]: So you are making this the default for conda? That means by default, we can't easily profile conda-installed libcudf. That concerns me a bit. Thoughts @kkraus14 ?

Comment (harrism) [reply conda/recipes/libcudf/build.sh]: If we *do* decide to make NVTX off by default, I recommend just changing the default in the cmake and adding a build script option to turn it ON.

File: build.sh (hunk 1 above)
Comment (jdye64) [reply build.sh]: You are right. That was not a smart flag name. Will change that. Not sure what I was thinking?

File: conda/recipes/libcudf/build.sh (hunk 2 above)
Comment (jdye64) [reply conda/recipes/libcudf/build.sh]: Thinking about this a little more I agree with you here. I'm going to remove the disabling of nvtx as the default. I'll make that change.

File: ci/local/README.md [hunk 3]
```diff
@@ -12,7 +12,7 @@ nvidia-docker
 ## Usage
 
 ```
-bash build.sh [-h] [-H] [-s] [-r <repo_dir>] [-i <image_name>]
+bash build.sh [-h] [-H] [-s] [-x] [-r <repo_dir>] [-i <image_name>]
```
Comment (dillon-cullinan) [inline ci/local/README.md]: This is a README for a different unrelated script. Should revert this.

File: conda/recipes/libcudf/build.sh (hunk 2 above)
Comment (dillon-cullinan) [reply conda/recipes/libcudf/build.sh]: Agreed with Mark, I would not disable NVTX in the conda build.

File: conda/recipes/libnvstrings/build.sh [hunk 4]
```diff
@@ -7,4 +7,4 @@ printenv
 # Cleanup local git
 git clean -xdf
 # build libnvstrings with verbose output
-./build.sh -v libnvstrings --allgpuarch
+./build.sh -v -x libnvstrings --allgpuarch
```
Comment (dillon-cullinan) [inline conda/recipes/libnvstrings/build.sh]: And here.

File: build.sh (hunk 1 above)
Comment (harrism) [reply build.sh]: Once you change this I can approve.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,6 +1,7 @@
 # cuDF 0.11.0 (Date TBD)
 
 ## New Features
+- PR #3525 Altered build.sh accept flag to disable nvtx
```
Comment (harrism) [inline CHANGELOG.md]:
```suggestion
- PR #3525 build.sh option to disable nvtx
```

File: build.sh [hunk 6]
```diff
@@ -18,21 +18,22 @@ ARGS=$*
...
+   libnvstrings     - build the nvstrings C++ code only
+   nvstrings        - build the nvstrings Python package
+   libcudf          - build the cudf C++ code only
+   cudf             - build the cudf Python package
+   dask_cudf        - build the dask_cudf Python package
+   benchmarks       - build benchmarks
+   -v               - verbose build mode
+   -g               - build for debug
+   -n               - no install step
+   --disable_nvtx   - do not build nvtx
```
Comment (harrism) [inline build.sh]:
```suggestion
   --disable_nvtx   - Disable inserting NVTX profiling ranges
```

Comment (jdye64) [reply build.sh]: no problem, just made the change

**Comment by jdye64** [comment]
rerun tests

**Comment by jdye64** [comment]
I'm going to place it in 0.13 (changelog) since I don't think its critical to have in 0.12 so no need to push it.

**Comment by raydouglass** [comment]
> I'm going to place it in 0.13 (changelog) since I don't think its critical to have in 0.12 so no need to push it.

Don't forget to retarget to `branch-0.13`

**Comment by kkraus14** [comment]
rerun tests

File: build.sh [hunk 7]
```diff
@@ -18,21 +18,22 @@ ARGS=$*
...
-   libcudf      - build the cudf C++ code only
-   cudf         - build the cudf Python package
-   dask_cudf    - build the dask_cudf Python package
-   benchmarks   - build benchmarks
-   -v           - verbose build mode
-   -g           - build for debug
-   -n           - no install step
-   --allgpuarch - build for all supported GPU architectures
-   -h           - print this text
+VALIDARGS="clean libnvstrings nvstrings libcudf cudf dask_cudf benchmarks -v -g -n -x --allgpuarch -h"
```
Comment (dillon-cullinan) [inline build.sh]: Add `--disable_nvtx` to the `VALIDARGS` variable.

Comment (kkraus14) [reply build.sh]: Done

**Comment by harrism** [comment]
rerun tests

**Pull request merged**